src/python/pants/backend/python/typecheck/mypy/subsystem.py

```diff
@@ -70,7 +70,10 @@ class MyPy(PythonToolBase):
     help_short = "The MyPy Python type checker (http://mypy-lang.org/)."
 
     default_main = ConsoleScript("mypy")
-    default_requirements = ["mypy>=0.961,<2"]
+    # Temporary upper bounds until one or both of:
+    # https://github.com/pantsbuild/pants/pull/21158
+    # https://github.com/pantsbuild/pants/pull/21387
+    default_requirements = ["mypy[python2]>=0.961,<1.5"]
```

Comment:
Test around py2.7 assumes `typed-ast` is present, which used to be implicit with 3.7.